server/src/com/thoughtworks/go/config/update/TemplateConfigCommand.java

```diff
@@ -46,7 +46,7 @@ public TemplateConfigCommand(PipelineTemplateConfig templateConfig, LocalizedOpe
     public boolean isValid(CruiseConfig preprocessedConfig) {
         TemplatesConfig templatesConfig = preprocessedConfig.getTemplates();
         preprocessedTemplateConfig = findAddedTemplate(preprocessedConfig);
-        preprocessedTemplateConfig.validate(ConfigSaveValidationContext.forChain(preprocessedConfig));
+        preprocessedTemplateConfig.validate(ConfigSaveValidationContext.forChain(templatesConfig));
```

Comment:
Not really introduced as a part of this PR. But looks like we do not seem to be validating things like fetch artifacts task for the template config api. This could lead to invalid config being saved. Not sure what kind of issues would this lead to.